### ISSUE TITLE
[1.28.1] cups-browsed gets SIGKILL during service restart

### ISSUE THREAD

**Issue by zdohnal** [OpenPrinting/cups-filters#298]
Hi,

we have a CI test which uses cups-browsed - it is compiled to don't keep generated queues and use RemoteName for created local queues pointing to remote cups queues.
Then I set ```BrowsePoll <cups-1.4.2 server>``` (I used ```BrowseFilter``` to simplify things too), wait for queues to show up and then stop the service. Stopping the service takes longer than with 1.28.0 and the service is killed (log from systemctl):

[Debug log](https://github.com/OpenPrinting/cups-filters/files/5134785/loading_queues_log.txt)
 for loading one queue and 
[debug log](https://github.com/OpenPrinting/cups-filters/files/5134795/stopping_service_log.txt) when stopping the service (it takes 1m 30s to stop the service - the SIGTERM signal times out and SIGKILL is sent) are attached.

IMO the issue is the signal is ignored and one commit in 1.28.1 moved signal handlers, so I'll try if reverting that commit will help.

**Comment by zdohnal** [comment]
My guess is correct, reverting the [commit](https://github.com/OpenPrinting/cups-filters/commit/ee8ef028c0fae4b570594166d00ce866fa69f864) which moved signal handling into ```get_printer_attributes5()``` fixes the issue.

**Comment by zdohnal** [comment]
@tillkamppeter signal handler with ```cancel_job()``` masks general ```sigterm_handler()``` so ```cups-browsed``` never shutdown itself if signal handler is in ```get_printer_attributes5()``` (probably because it is loaded in ```cups-browsed``` binary).
If the signal handler is within ```get_printer_attributes4()```, then it is called only from ```driverless/driverless-fax``` executables, so ```cups-browsed``` is not affected.

**Comment by tillkamppeter** [comment]
Fixed in commit 23a9dc4010 on master. Thank you very much for your report.
The signal handlers got in with the student work for IPP Fax Out support but they are unneeded. converting DNS-SD-based URIs into standard IPP URIs and polling printer attributes are rather quick operations and there are also no problems with unclean shutdown as one had with a print job. So signal handling is overkill here and therefore I removed it.
